### PR TITLE
Adds notes and orders worksets by date run

### PR DESCRIPTION
**Pull request by Galithil**

### CODE REVIEW

File: run_dir/design/workset_samples.html [hunk 1]
```diff
@@ -13,27 +13,78 @@
 <div id="page_content" style="display:none;">
 
     <h1>Workset <span id="page_title">{{ workset_name}}</span></h1>
-    <button type="button" class="btn btn-primary btn-lg" data-toggle="modal" data-target="#modal_plate">
+    <h3>Lims Step : <span id="span_lims_step"></span>, run on <span id="date_run"></span></h3>
```
Comment (ewels) [inline run_dir/design/workset_samples.html]: For these empty tags where we will be inserting data, I like to fill the tag with some holding content in the template. This makes it much more obvious when it has failed to load. Just a hyphen or question mark is fine. So for this case I would use:

```html
 <h3>Lims Step : <span id="span_lims_step">?</span>, run on <span id="date_run">?</span></h3>
```

File: run_dir/design/workset_samples.html [hunk 2]
```diff
@@ -13,27 +13,78 @@
...
-      </button>
-    <div id="workset_tables">
+        </button>
     </div>
-    <div id="modal_plate" class="modal fade" >
-        <div class="plate_modal modal-dialog"> 
-             <div class="modal-content">
+    <div class="tabbable">
+        <ul class="nav nav-tabs">
+        <li class="active"><a href="#workset_tables" role="tab" data-toggle="tab">Workset Tables</a></li>
```
Comment (ewels) [inline run_dir/design/workset_samples.html]: Your lack of nested whitespace tabbing is making me sad :disappointed:

File: run_dir/design/workset_samples.html [hunk 3]
```diff
@@ -13,27 +13,78 @@
...
+          <div class="panel-group" id="workset_notes"></div>
+        </div>
+        </div>
+        <div id="modal_plate" class="modal fade" >
+            <div class="plate_modal modal-dialog"> 
+                 <div class="modal-content">
                 <input type="button" Onclick="updatePlate('rec_ctrl')" value=" Reception Control " />
                 <input type="button" Onclick="updatePlate('library')" value=" Library Validation " />
                 <input type="button" Onclick="updatePlate('seq')" value=" Sequencing" />
                 <canvas id="myCanvas" width="1278" height="860" style="border:1px solid #000000;">
```
Comment (ewels) [inline run_dir/design/workset_samples.html]: What's this element for?

**Comment by ewels** [comment]
Looks good - is it on stage for testing?

My only problem with this is that it seems to have lots of copied and pasted code. Is it possible to move this stuff into a more general javascript file somewhere and then have both pages load that? It would make maintenance easier in the future if there's only one piece of code to update...

File: run_dir/design/workset_samples.html (hunk 2 above)
Comment (Galithil) [inline run_dir/design/workset_samples.html]: it's the same line :D

File: run_dir/design/workset_samples.html (hunk 3 above)
Comment (Galithil) [inline run_dir/design/workset_samples.html]: the plate view

File: run_dir/design/workset_samples.html (hunk 2 above)
Comment (ewels) [inline run_dir/design/workset_samples.html]: Not from the `<ul>`?

File: run_dir/design/workset_samples.html (hunk 3 above)
Comment (ewels) [inline run_dir/design/workset_samples.html]: Fair enough - I just couldn't find `myCanvas` referred to anywhere else in the code so I was confused..

Comment (Galithil) [inline run_dir/design/workset_samples.html]: should be in the js

**Comment by Galithil** [comment]
I am not sure that they actually can be reunited.
I changed how the workset notes are handled because the way running notes are added does not work with my wish to be able to delete them : When adding a running note, there's no actual refresh of the running notes, but rather a new element to the current js object is added. That avoids one server call, but at the same time, that messes up the timesamp on it by a few milliseconds, which means that I can't delete the note because the id just changed. 
The main table is built on the fly, with column names in js, so that's pretty specific.
Most of the js code has slight changes, the only two functions that have been left untouched are "make project links" and the markdown thing.

**Comment by ewels** [comment]
Ok, fair enough.. Might be nice to unite just the markdown stuff into a `gs-markdown.js` file or something? But only if you can be bothered. Otherwise I'm happy to merge. Let me know.

**Comment by Galithil** [comment]
Was in uppsala today, and i'll be at the hadoop thingy tomorrow and friday,
so ...

**Comment by guillermo-carrasco** [comment]
ping @Galithil @ewels :-)

**Comment by Galithil** [comment]
yeah this should be mergeable.  I don't really want to deploy it on dev because the DB is not up to date, and I can't make it up to date because the lims is not up to date either :D

**Pull request merged**